Security updates

lam00071 opened the pull request:
This pull request will have e2e, karma test and the implementation to prevent word list name contains special character/punctuation that causing the GET request method to be failed. Closes #33 #63 #64

lam00071 committed:
Write e2e tests for not allowing special character or punctuation in wordlist name 

This should have the e2e test to not allow users to have special characters or punctuation in the wordlist name. Issue #63

Co-authored-by: Richard Lussier <lussi036@morris.umn.edu>
Co-authored-by:  Joshua Eklund <eklun124@umn.edu>
Co-authored-by:  Jacob Jenness <jenne077@morris.umn.edu>
Co-authored-by:  Elena Lam <lam00071@morris.umn.edu>
Co-authored-by:  Biruk Mengistu <mengi024@morris.umn.edu>

lam00071 committed:
Write karma tests for not allowing users to have special characters or punctuation in the wordlist name

This should have karma tests to not allow users to have special characters or punctuation in the wordlist name. Issue #64

Co-authored-by: Richard Lussier <lussi036@morris.umn.edu>
Co-authored-by:  Joshua Eklund <eklun124@umn.edu>
Co-authored-by:  Jacob Jenness <jenne077@morris.umn.edu>
Co-authored-by:  Elena Lam <lam00071@morris.umn.edu>
Co-authored-by:  Biruk Mengistu <mengi024@morris.umn.edu>

lam00071 committed:
Implement feature to prevent wordlist name contains special character or punctuation

This issue will fix the bug when wordlist name has special character/punctuation which makes GET request failed, causing exporting and deleting method not working. Issue #33

Co-authored-by: Richard Lussier <lussi036@morris.umn.edu>
Co-authored-by:  Joshua Eklund <eklun124@umn.edu>
Co-authored-by:  Jacob Jenness <jenne077@morris.umn.edu>
Co-authored-by:  Elena Lam <lam00071@morris.umn.edu>
Co-authored-by:  Biruk Mengistu <mengi024@morris.umn.edu>

RichardLussier reviewed (changes requested):
Add a change so "-" is accepted by our regular expression

lam00071 committed:
Allow user to use a dash (-) in word list name

This commit will allow user to use a dash when they create a word list. It will also have karma test for this.

Co-authored-by: Richard Lussier <lussi036@morris.umn.edu>
Co-authored-by:  Joshua Eklund <eklun124@umn.edu>
Co-authored-by:  Jacob Jenness <jenne077@morris.umn.edu>
Co-authored-by:  Elena Lam <lam00071@morris.umn.edu>
Co-authored-by:  Biruk Mengistu <mengi024@morris.umn.edu>

biruk741 reviewed (approved):
Hmm... good stuff.

RichardLussier reviewed (approved):
Looks good
